classes/local/store/s3/client.php

```diff
@@ -571,26 +571,32 @@ public function get_expiration_time($now, $expires) {
             // Convert to a valid timestamp.
             $expires = strtotime($expires);
         }
+        // Invalid or already expired:
         // If it's set to 0 or strtotime() returned false,
         // set it to default + 1 min as a healthy margin.
         if (empty($expires)) {
             $expires = $now + $this->expirationtime + MINSECS;
         }
+        // Expiry too short, push it out to the next 2 minutes (will round down later):
         // If it's already expired or expires less than in a minute,
-        // set it to 1 minute.
+        // set it to 2 minutes. This works together with rounding down later on
+        // to ensure a non zero expiry time, and a minimum expiry of 1 minute.
         if ($expires < $now + MINSECS) {
-            $expires = $now + MINSECS;
+            $expires = $now + (2 * MINSECS);
         }
+        // Checks upper bound:
         // The expiration date of a signature version 4 presigned URL must be
         // less than one week. So if it's greater than a week set it to 1 week.
         // Use MINSECS as a healthy margin of error.
         if ($expires - $now > WEEKSECS - MINSECS) {
             $expires = $now + WEEKSECS - MINSECS;
         }
-        if (is_null($expires) || false === $expires) {
-            // Invalid date format use default instead.
-            $expires = $now + $this->expirationtime;
-        }
+        // Rounds (down) to nearest minute:
+        // With our new expiry time, ensure we round down to the nearest minute
+        // (#457) to ensure expiry of potentially the same file will use the
+        // same URL, and will result in less duplicate requests.
+        $expires -= ($expires % MINSECS);
```

Comment:
All the logic looks good, can you amend the data provider here:

https://github.com/catalyst/moodle-tool_objectfs/blob/a43206024972ef1e61413e620971e90808844922/tests/object_file_system_test.php#L714